sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/Authentication.java

```diff
@@ -81,53 +81,60 @@ private static void authenticationWithAzureActiveDirectoryFormTrainingClient() {
-                    transactionDateField.getConfidence());
+                    transactionDate, transactionDateField.getConfidence());
             }
-            FormField receiptItemsField = recognizedFields.get("Items");
-            System.out.printf("Receipt Items: %n");
-            if (receiptItemsField.getFieldValue().getType() == FieldValueType.LIST) {
-                List<FormField> receiptItems = receiptItemsField.getFieldValue().asList();
-                receiptItems.forEach(receiptItem -> {
-                    if (receiptItem.getFieldValue().getType() == FieldValueType.MAP) {
-                        receiptItem.getFieldValue().asMap().forEach((key, formField) -> {
-                            if (key.equals("Name")) {
-                                if (formField.getFieldValue().getType() == FieldValueType.STRING) {
-                                    System.out.printf("Name: %s, confidence: %.2fs%n",
-                                        formField.getFieldValue().asString(),
-                                        formField.getConfidence());
+
+            FormField<?> receiptItemsField = recognizedFields.get("Items");
+            if (receiptItemsField != null) {
+                System.out.printf("Receipt Items: %n");
+                if (receiptItemsField.getFieldValue() instanceof List) {
+                    List<FormField<?>> receiptItems = (List<FormField<?>>) receiptItemsField.getFieldValue();
+                    receiptItems.forEach(receiptItem -> {
+                        if (receiptItem.getFieldValue() instanceof Map) {
+                            ((Map<String, FormField<?>>) receiptItem.getFieldValue()).forEach((key, formField) -> {
+                                if ("Name".equals(key)) {
+                                    if (formField.getFieldValue() instanceof String) {
+                                        System.out.printf("Name: %s, confidence: %.2fs%n",
+                                            formField.getFieldValue(),
+                                            formField.getConfidence());
+                                    }
                                 }
-                            }
-                            if (key.equals("Quantity")) {
-                                if (formField.getFieldValue().getType() == FieldValueType.INTEGER) {
-                                    System.out.printf("Quantity: %s, confidence: %.2f%n",
-                                        formField.getFieldValue().asInteger(), formField.getConfidence());
+                                if ("Quantity".equals(key)) {
+                                    if (formField.getFieldValue() instanceof Integer) {
+                                        System.out.printf("Quantity: %d, confidence: %.2f%n",
+                                            formField.getFieldValue(), formField.getConfidence());
+                                    }
```

Comment:
Since you are checking `instanceof` here, it's better to show that you are converting it into `Integer` type by casting it: `Integer quantity = (Integer) formField.getFieldValue()`. The `printf` statement with `%d` is too subtle to understand why you checked if this was an `instanceof` an Integer. 

Make the same changes in all places where you are checking for `instanceof`.